dhall/src/Dhall.hs

```diff
@@ -1860,6 +1857,34 @@ instance Inject a => Inject (Data.HashSet.HashSet a) where
 
 instance (Inject a, Inject b) => Inject (a, b)
 
+{-| Inject a `Data.Map` to a @Prelude.Map.Type@
+
+>>> prettyExpr $ embed inject (Data.Map.fromList [(1 :: Integer, True )])
+[ { mapKey = +1, mapValue = True } ]
+
+>>> prettyExpr $ embed inject (Data.Map.fromList [] :: Data.Map.Map Integer Bool)
+[] : List { mapKey : Integer, mapValue : Bool }
+
+-}
+instance (Inject k, Inject v) => Inject (Data.Map.Map k v) where
+    injectWith options = InputType embedOut declaredOut
+      where
+        embedOut m = ListLit listType $ mapEntries m
+          where
+            listType
+                | Data.Map.null m = Just declaredOut
+                | otherwise       = Nothing
+
+        declaredOut = App List $ Record $ Dhall.Map.fromList
```

Comment:
So far the code base mostly uses parentheses instead of `$`. @Gabriel439 can probably explain why.

I think we should stick with the convention unless there's a good reason not to.

Reply:
It comes from this: http://www.haskellforall.com/2015/09/how-to-make-your-haskell-code-more.html

... although I usually only apply that to my own coding style.  I usually do not review other people's code for style that much

Reply:
Interesting read, thank you!